usr/share/rear/rescue/GNU/Linux/310_network_devices.sh

```diff
@@ -854,10 +859,17 @@ function handle_physdev () {
 
     DebugPrint "$network_interface is a physical device"
 
-    if [ -e $sysfspath/bonding_slave/perm_hwaddr ] ; then
-        mac="$( cat $sysfspath/bonding_slave/perm_hwaddr )"
-    else
-        mac="$( cat $sysfspath/address )" || BugError "Could not read a MAC address for '$network_interface'."
+    local mac=""
+
+    if has_binary ethtool ; then
+        mac="$( ethtool -P $network_interface 2>/dev/null | awk '{ print $3 }' )"
```

Comment:
Seems fragile to rely on the MAC address being the third field of ethtool output. What about taking the last field?
```
awk '{print $NF}'
```
(Of course parsing this output which was obviously not designed for anything else than for being seen by a human is horribly barbaric. Why can't ethtool have a more script-friendly output, or why can't sysfs expose this information?)

Reply:
+1

Reply:
> Why can't ethtool have a more script-friendly output ...

Was asking similar question with `xfs_info` some time ago :-)

V.